internal/project/project.go

```diff
@@ -372,14 +373,23 @@ func (p *Project) getScriptKind(fileName string) core.ScriptKind {
 }
 
 func (p *Project) markFileAsDirty(path tspath.Path) {
-	p.markAsDirty()
+	p.dirtyStateMu.Lock()
+	defer p.dirtyStateMu.Unlock()
+	if !p.dirty {
+		p.dirty = true
+		p.dirtyFilePath = path
+		p.version++
+	} else if path != p.dirtyFilePath {
+		p.dirtyFilePath = ""
+	}
 }
 
 func (p *Project) markAsDirty() {
 	p.dirtyStateMu.Lock()
 	defer p.dirtyStateMu.Unlock()
 	if !p.dirty {
 		p.dirty = true
+		p.dirtyFilePath = ""
```

Comment:
I think we need to reset `p.dirtyFilePath` unconditionally here. Suppose we call `p.markFileAsDirty()` and it sets `p.dirtyFilePath` to some path, and then right after we call `p.markAsDirty()`: we won't reset `p.dirtyFilePath` because `p.dirty` is already true.

Reply:
Good catch!